### DISCUSSION TITLE
interface conditions

### DISCUSSION THREAD

**Discussion by dealmeidavf** [idaholab/moose#17828] [Q&A General] [answered]
Hello,

Is there a MOOSE documentation of a weak formulation involving interfaces that I can read about? I am interested in understanding, at the mathematical level, what MOOSE does when we code a class derived from `InterfaceKernel`. I believe MOOSE considers unknown variables in different subdomains as discontinuous across the interface. If the interface condition calls for continuity, this needs to be imposed weakly using an `InterfaceKernel` class. Is there another way to impose continuity at an interface?

A related issue is the code:

https://github.com/idaholab/moose/blob/1585ebf7097cebab2ccfc47358ac20ca572ce847/modules/heat_conduction/src/interfacekernels/ConjugateHeatTransfer.C#L42-L56

If the temperature is continuous at the interface, this would make the return value of the residual equal to zero correct?

Thanks.

--
Valmor

**Comment by jiangwen84** [comment, marked as answer]
`InterfaceKernel` allows you to implement a weak form at the interface by providing you the shape/test functions, variable values, etc at both sides of the interface. At the mathematical level, you need to derive your weak form baed on your interfacial condition. It can be used to enforce continuity condition, like `PenaltyInterfaceDiffusion.C`, or transfer heat from one domain to another, like `ConjugateHeatTransfer.C`. 

For your second question, if the temperature is continuous, the residual will be zero. Physically, it means there is no heat transfer if two domains have the same temperature across the interface.

**Reply by dealmeidavf** [reply]
From this signature:

> Real 
 ConjugateHeatTransfer::computeQpResidual(Moose::DGResidualType type) 
 { 

I take it the test function of the interface condition is discontinuous on finite elements. Correct?

Also on your last comment, even with continuous temperature there may still exist a non-zero heat flux; hence heat transfer can occur. The gradient of temperature at the interface is the deciding factor.

I don't seem to be able to find a Galerkin weak formulation for interfacial problems in the MOOSE site/distribution. It would be instructive to have one or be directed to one.

Thanks.

**Reply by jiangwen84** [reply]
`ConjugateHeatTransfer` is not used to enforce temperature continuity across the interface unless the heat transfer coefficient is large enough (then it becomes the penalty parameter). It actually applies a Neumann boundary condition to both sides of the interface where the flux is defined as `_h[_qp]) * (_T_fluid[_qp] - _neighbor_value[_qp])`. The flux is always the same but opposite at two sides of the interface, so the flux is continuous. When there is no temperature jump, the flux becomes zero and it would apply a zero flux BC.

Sounds like this might not be what you would like to use for your application. If you could let us know what interfacial conditions you would like to apply, we can help on the weak form derivation and implementation. 

For your first question, I believe `interfaceKernel` can be used on both continuous or discontinuous interface. It depends on whether your mesh is continuous or not across the interface.
